doc/services/pm/device.rst

```diff
@@ -158,8 +158,47 @@ a particular device. It is important to emphasize that, although the
 state is tracked by the subsystem, it is the responsibility of each device driver
 to handle device actions(:c:enum:`pm_device_action`) which change device state.
 
-Each :c:enum:`pm_device_action` have a direct an unambiguous relationship with
-a :c:enum:`pm_device_state`.
+Each :c:enum:`pm_device_state` state is defined as follows:
+
+- ``OFF``: Device hardware is not powered. This is the initial state from which
+  a device driver is initialized.
+
+  When a device driver is initialized, we do not know the state of the device.
+  As a result, the :c:enum:`pm_device_action` ``TURN_ON`` action should be able
+  to transition the device from any internal state into ``SUSPENDED``, since no
+  guarantees can be made across resets. This is typically achieved through
+  toggling a reset pin or triggering a software reset through a register write
+  before performing any additional configuration needed to meet the requirements
+  of ``SUSPENDED``. For devices where this is not possible, the device driver
+  must presume the device is in either the ``OFF`` or ``SUSPENDED`` state at
+  time of initialization, as these are the states within which device drivers
+  may be deinitialized.
+
+- ``SUSPENDED``: Device hardware is powered, but the device is not needed by the
+  system. The device should be put into its lowest internal power state,
+  commonly named "disabled" or "stopped". The device will be powered off if its
+  power domain is no longer needed by the system. Note that this state is
+  **NOT** a "low-power"/"partially operable" state, those are configured using
+  device driver specific APIs, and apply only while the device is ``ACTIVE``.
+
+  A device driver may be deinitialized in this state. Once the device driver
+  has been deinitialized, we implicitly move to the ``OFF`` state as the device
+  hardware may lose power, with no device driver to respond to the
+  corresponding :c:enum:`pm_device_action` ``TURN_OFF`` action.
+
+- ``ACTIVE``: Device hardware is powered, and the device is needed by the
+  system. The device should be enabled in this state. Any device driver API may
+  be called in this state.
+
+- ``SUSPENDING``: Device hardware is powered, but the device has been enqueued to
+  be suspended, as it is no longer needed by the system. The device will be
+  dequeued in case the device becomes needed by the system, otherwise the device
+  will be suspended. No device driver API calls must occur in this state. Note
+  that this state is opaque to the device driver (no :c:enum:`pm_device_action`
+  is called as this state is entered)
```

Comment:
this doesn't sound applicable to all systems, should an example be explained? why would there be a queue? this is something I actually never understood and neither has anyone I talked to about it so maybe an explanation of why this state exists, somewhere, would be good

Reply:
added the _why_ and `when` :)

Reply:
This is very useful to avoid unnecessary power cycles for a device. Depending on who is getting a device it may not know if it will be used soon (imagine a subsystem) . If you are asking strictly about the state (and not the functionality) is pretty much because it is needed to implement the functionality described.